Dockerfile

```diff
@@ -4,34 +4,35 @@ FROM golang:alpine as builder
 # queries required to connect to linked containers succeed.
 ENV GODEBUG netdns=cgo
 
-# Install dependencies and build the binaries.
+# Copy in the local repository to build from.
+COPY . /go/src/github.com/lightningnetwork/lnd
+
+# Install dependencies.
 RUN apk add --no-cache --update alpine-sdk \
     git \
-    make \
-    gcc \
-&&  git clone https://github.com/lightningnetwork/lnd /go/src/github.com/lightningnetwork/lnd \
-&&  cd /go/src/github.com/lightningnetwork/lnd \
-&&  make \
-&&  make install
+    make
+
+# Build the binaries.
+WORKDIR /go/src/github.com/lightningnetwork/lnd
+RUN make install
 
 # Start a new, final image.
 FROM alpine as final
 
 # Define a root volume for data persistence.
 VOLUME /root/.lnd
 
-# Add bash and ca-certs, for quality of life and SSL-related reasons.
-RUN apk --no-cache add \
-    bash \
-    ca-certificates
```

Comment:
Why remove `ca-certificates` here?

Reply:
Because it's useless, AFAIK.

The same goes for bash (although it's debatable since there is [a completion script for bash](https://github.com/lightningnetwork/lnd/blob/master/contrib/lncli.bash-completion) that could be installed in this image).

Reply:
Is it useless because of the copy change you made, or just useless in general.  Maybe this could be part of a separate cleanup PR?

Reply:
Because there is nothing that relies on CA certificates in this container (again AFAIK).
Anyway, most of the CA certs are already included in the (installed by default) [`ca-certificates-cacert` package](https://pkgs.alpinelinux.org/package/v3.9/main/x86_64/ca-certificates-cacert).

Reply:
I think this is only needed in "the other" Dockerfile because [certificates](https://github.com/lightningnetwork/lnd/blob/master/docker/docker-compose.yml#L93) are used to secure communication between services.  I'd be in favor of removing as a different PR if it's not being used in this Dockerfile.

Reply:
It seems like you might be right. It's possible to add TLS to some btcd RPC api for example. I've got no idea how and if it would work with trusted certificatition authority --- ie. will they issue a certificate for such a service? is lnd going to check that certificates against the installed ca certificates?

I'll remove that change for now and double check if that's a needed / useful package. If not, I will create a new PR.

Reply:
It looks like it's just a self-sighned cert (no certificate authority) such that if both lnd and btcr use the same cert than the authentication is successful:

lnd: https://github.com/lightningnetwork/lnd/blob/master/docker/lnd/start-lnd.sh#L58
btcr cert: https://github.com/lightningnetwork/lnd/blob/master/docker/btcd/start-btcd.sh#L58
btcr key: https://github.com/lightningnetwork/lnd/blob/master/docker/btcd/start-btcd.sh#L59

That package would be needed if `/Dockerfile` was connecting to a node that required it